packages/workspace/src/tasks-runner/run-command/components/run-command.tsx

```diff
@@ -41,52 +40,63 @@ export function RunCommandComponent({
       projectNames: params.projectNames,
       tasks: null,
       taskResults: null,
+      tasksToTerminalOutputs: {},
     });
     setRunCommandState('RUN_MANY');
   };
 
-  callbacks.onStartTasks = (tasks) => {
+  lifeCycle.callbacks.onStartTasks = (tasks) => {
     setTasksState((state) => ({
       ...state,
       tasks: [...(state.tasks || []), ...tasks],
     }));
   };
 
-  callbacks.onEndTasks = (taskResults) => {
+  lifeCycle.callbacks.onEndTasks = (taskResults) => {
     setTasksState((state) => ({
       ...state,
       taskResults: [...(state.taskResults || []), ...taskResults],
     }));
   };
 
+  lifeCycle.callbacks.onPrintTaskTerminalOutput = (
+    task,
+    _cacheStatus,
+    output
+  ) => {
+    setTasksState((state) => ({
+      ...state,
+      tasksToTerminalOutputs: {
+        ...state.tasksToTerminalOutputs,
+        [task.id]: output,
+      },
+    }));
+  };
+
+  function clearTaskOutputCacheForTaskId(taskId: string) {
```

Comment:
This is likely overkill for most tasks, but figured there will be some custom ones out there (we have one on client work for example) that have a very large amount of output and so it probably makes sense to discard it when it's no longer needed